src/ElementalChat.vue

```diff
@@ -215,30 +215,40 @@ export default {
     handleToDisplay () {
       return this.isHoloAnonymous ? 'anonymous' : this.agentHandle
     },
-    shouldAutoSignIn () {
-      // Note: isAnonymousEnabled refers to whether the feature is enabled when the app is built. isHoloAnonymous refers to the current status of our hosted agent.
-      return isHoloHosted() && !isAnonymousEnabled() && this.holoStatus === 'ready' && this.isHoloAnonymous
+    shouldHandleLogin() {
+      return isHoloHosted() && this.holoStatus === 'ready' && this.isHoloAnonymous === true
     },
     canMakeZomeCalls() {
+      // Note: isAnonymousEnabled refers to whether the feature is enabled when the app is built. isHoloAnonymous refers to the current status of our hosted agent.
       return isHoloHosted() ? this.holoStatus === 'ready' && (isAnonymousEnabled() || this.isHoloAnonymous === false) : !this.conductorDisconnected
     }
   },
   created () {
     console.log('agentKey started as', this.agentKey)
   },
   watch: {
-    async shouldAutoSignIn (should) {
+    async shouldHandleLogin (should) {
+      console.log(`watcher activated: shouldHandleLogin=${should}`)
       if (should) {
-        await this.holoLogin()
+        const urlParams = new URLSearchParams(window.location.search)
+        if (urlParams.has('signin')) {
+          this.$store.dispatch('holochain/holoSignin')
+        } else if (urlParams.has('signup') || !isAnonymousEnabled()) { 
+          this.$store.dispatch('holochain/holoSignup')
+        }
       }
     },
```

Comment:
This is missing the piece that automatically calls sign in if we're anonymous and `!isAnonymousEnabled()`

Reply:
Oh nevermind. I was just blind to it on line 235